### ISSUE TITLE
Failed to build `ghc-paths` on macOS

### ISSUE THREAD

**Issue by cjwcommuny** [simonmar/ghc-paths#28]
Tried to build `ghc-paths` on macOS 13.0 (M1 Mac), but got the following error:

```
$ cabal build
Build profile: -w ghc-9.2.2 -O1
In order, the following will be built (use -v for more details):
 - ghc-paths-0.1.0.12 (lib:ghc-paths) (first run)
cabal: Failed to build ghc-paths-0.1.0.12. The failure occurred during the
configure step. The build process segfaulted (i.e. SIGSEGV).
```

## Other info

```
$ cabal --version
cabal-install version 3.6.2.0
compiled using version 3.6.2.0 of the Cabal library
```

```
$ ghc --version
The Glorious Glasgow Haskell Compilation System, version 9.2.2
```

**Comment by Bodigrim** [comment]
Same issue, macOS 13.0 (M2), cabal 3.8.1.0, GHC 9.2.4.

**Comment by Bodigrim** [comment]
This seems to be a manifestation of https://gitlab.haskell.org/ghc/ghc/-/issues/21964, fixed in GHC 9.4 series. This issue can be closed, it has nothing to do with `ghc-paths` per se.
